testing/trino-product-tests/src/main/java/io/trino/tests/product/TestImpersonation.java

```diff
@@ -47,24 +51,77 @@ public class TestImpersonation
     @Named("databases.alice@presto.jdbc_user")
     private String aliceJdbcUser;
 
+    @Inject
+    @Named("databases.bob@presto.jdbc_user")
+    private String bobJdbcUser;
+
     // The value for configuredHdfsUser is profile dependent
     // For non-Kerberos environments this variable will be equal to -DHADOOP_USER_NAME as set in jvm.config
     // For Kerberized environments this variable will be equal to the hive.hdfs.trino.principal property as set in hive.properties
     @Inject
     @Named("databases.presto.configured_hdfs_user")
     private String configuredHdfsUser;
 
+    @Inject
+    @Named("databases.hive.warehouse_directory_path")
+    private String warehouseLocation;
+
     @BeforeMethodWithContext
     public void setup()
     {
         aliceExecutor = connectToTrino("alice@presto");
+        bobExecutor = connectToTrino("bob@presto");
     }
 
     @AfterMethodWithContext
     public void cleanup()
     {
         // should not be closed, this would close a shared, global QueryExecutor
         aliceExecutor = null;
+        bobExecutor = null;
+    }
+
+    @Test(groups = {HDFS_IMPERSONATION, PROFILE_SPECIFIC_TESTS})
+    public void testExternalLocationTableCreationFailure()
+    {
+        String commonExternalLocationPath = warehouseLocation + "/nested_" + randomNameSuffix();
+
+        String tableNameBob = "bob_external_table";
+        String tableLocationBob = commonExternalLocationPath + "/bob_table";
+        bobExecutor.executeQuery(format("CREATE TABLE %s (a bigint) WITH (external_location = '%s')", tableNameBob, tableLocationBob));
+        String owner = hdfsClient.getOwner(commonExternalLocationPath);
+        assertEquals(owner, bobJdbcUser);
+
+        String tableNameAlice = "alice_external_table";
+        String tableLocationAlice = commonExternalLocationPath + "/alice_table";
+        assertQueryFailure(() -> aliceExecutor.executeQuery(format("CREATE TABLE %s (a bigint) WITH (external_location = '%s')", tableNameAlice, tableLocationAlice)))
+                .cause()
+                .hasMessageContaining("Failed to create directory, permission denied");
+
+        bobExecutor.executeQuery(format("DROP TABLE IF EXISTS %s", tableNameBob));
+        aliceExecutor.executeQuery(format("DROP TABLE IF EXISTS %s", tableNameAlice));
+    }
+
+    @Test(groups = {HDFS_NO_IMPERSONATION, PROFILE_SPECIFIC_TESTS})
+    public void testExternalLocationTableCreationSuccess()
+    {
+        String commonExternalLocationPath = warehouseLocation + "/nested_" + randomNameSuffix();
+
+        String tableNameBob = "bob_external_table";
```

Comment:
This name is already used in another test. If `testExternalLocationTableCreationFailure` failed before removing the table, this test fails unexpectedly. The test method shouldn't affect each other. 

I would recommend using `randomNameSuffix()`.

Reply:
refactored